mitosheet/src/components/elements/MultiSelectButtons.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Mito
+import React from 'react';
+import { classNames } from '../../utils/classNames';
+import '../../../css/elements/MultiSelectButtons.css'
+
+interface MultiSelectButtonsProps {
+    values: string[];
+    selectedValues: string[];
+    onChange: (newValue: string) => void;
+}
+
+/**
+ * The MultiSelectButtons component displays a set of options and allows
+ * the user to select as many as they'd like
+ */ 
+const MultiSelectButtons = (props: MultiSelectButtonsProps): JSX.Element => {
```

Comment:
Ok, a few thoughts on this element.

We have learned over time that making the _value_ you need to display the same as the value you need in the callbacks adds a lot of boilerplate code. You can see this through the way that you naturally ended up storing the delimiter values. 

Instead, we normally allow you to separate an id from the title ( you can see this in the DropdownItems anywhere we use column ids). And note here, creating a MultiSelectButtonItem would be the natural way to do this well.

This leads to a more complex component design, but a lot simpler consumption of these components. Imagine the following code:

```
// NOTE: the delimiters are now stateless. E.g. there is no notion of included within the delimeter
// definition (which is very sensible, we want a definition of just what delimiters we support - not // what we support and if they are included in one variable).
const delimiters = [{',': 'Comma', ' ': 'Space'}]

interface SplitTextToColumnParams {
    sheet_index: number,
    column_id: ColumnID,
    delimiters: string[]
}

const {params, setParams} = useSendEditOnClick(...);


<MultiToggleButtonContainer onToggle={(newDelimiters) => {
     setParams(prevParams => {
         return {
             ...prevParams,
             delimiters: newDelimiters
         }
     })

}}>
     {Object.entries(delimiters).map(([delimiter, delimiterTitle]) => {
          return (
              <MultiToggleButtonItem
                   id={delimiter}
                   title={delimiterTitle}
                   toggled={params.delimiters.includes(delimiter)}
              />
          )
     })}
</MultiToggleButtonContainer>
<Input placeholder="other" onChange={(e) => {
    const newValue = e.target.value;
    setParams(prevParams => {
        const newDelimiters = [...prevParams.delimiters].filter(delimiter => Object.keys(delimiters.includes(delimiter));
        newDelimiters.push(newValue);
        return {
            ...prevParams,
            delimiters: newDelimiters
        }
    })
}}>
```
Of course, this is a sketch of the code I think would work - I'm sure you'll have to adapt to figure out how to do this best. Give me a call if you want to chat implementation.

This has a couple of really important benefits:
1. It removes more state from the taskpane except that defined through our new hook. This makes refactoring like _10x_ easier. E.g. for example look at my drop duplicates refactor, vs. the fact I gave up on the excel import refactor b/c it doesn't use the new hook, and has a bunch of random state!
2. It is much closer to the designs of our other components. This is a very similar interface to what we use elsewhere, which will make using this much more of a breeze.
3. It would lead to _more_ code in the components, and less in the taskpane. Whenever we can do this, it's a good idea!

I think these changes are important to do, for the above reasons. I know they are pretty big, so take it in pieces. Do not crush yourself trying to get these done. We can just push back the release date on this feature - marathon not a sprint!

Reply:
Done!